lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -2049,6 +2050,17 @@ SPIRVValue *LLVMToSPIRV::transIntrinsicInst(IntrinsicInst *II,
     return BM->addExtInst(STy, BM->getExtInstSetId(SPIRVEIS_OpenCL), ExtOp, Ops,
                           BB);
   }
+  case Intrinsic::abs: {
+    if (!checkTypeForSPIRVExtendedInstLowering(II, BM))
```

Comment:
I think `checkTypeForSPIRVExtendedInstLowering` should be extended to validate `Intrinsic::abs` parameters.